autoload/test/javascript.vim

```diff
@@ -4,7 +4,8 @@ let test#javascript#patterns = {
 \}
 
 function! test#javascript#has_package(package) abort
-  for line in readfile('package.json')
+  let json_path = findfile('package.json', '.;')
```

Comment:
We should get this to exclude node_modules as that will contain main package.json files